sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/http/client/DefaultHttpClient.java

```
@@ -192,39 +196,214 @@ private void sendBody(HttpRequest httpRequest, Object progressReporter, HttpURLC
+        StringBuilder eventData = new StringBuilder();
+        ServerSentEvent event = new ServerSentEvent();
+
+        //start parsing line by line
+        for (String line : lines) {
+            int idx = line.indexOf(':');
+            if (idx < 0) {
+                continue; // ignore invalid data
+            } else if (idx == 0) {
+                // capture comment line
+                event.setComment(line.substring(1).trim());
+                continue;
+            }
+            String field = line.substring(0, idx);
+            String value = line.substring(idx + 1).trim();
+            switch (field.trim().toLowerCase()) {
+                case "event":
+                    event.setEvent(value);
+                    continue;
+
+                case "data":
+                    if (eventData.length() > 0) {
+                        eventData.append("\n");
+                    }
+                    eventData.append(value);
+                    continue;
 
-            AccessibleByteArrayOutputStream outputStream = new AccessibleByteArrayOutputStream();
+                case "id":
+                    if (!value.isEmpty()) {
+                        event.setId(Long.parseLong(value));
+                    }
+                    continue;
 
-            try (InputStream errorStream = connection.getErrorStream();
-                 InputStream inputStream = (errorStream == null) ? connection.getInputStream() : errorStream) {
-                byte[] buffer = new byte[8192];
-                int length;
-                while ((length = inputStream.read(buffer)) != -1) {
-                    outputStream.write(buffer, 0, length);
+                case "retry":
+                    if (!value.isEmpty() && DIGITS_ONLY.matcher(value).matches()) {
+                        event.setRetryAfter(Long.parseLong(value));
+                    }
+                    continue;
+
+                default:
+                    continue;
```

We should throw here if we get invalid data.